### PR TITLE
adding tenantId field in model group index

### PR DESCRIPTION
**Pull request by dhrubo-os**
### Description
[Describe what this change achieves]

### Related Issues
Resolves #[Issue number to be closed when this PR is merged]

### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md).
- [ ] Commits are signed per the DCO using `--signoff`.
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/ml-commons/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: common/src/main/java/org/opensearch/ml/common/CommonValue.java [hunk 1]
```diff
@@ -58,7 +58,7 @@ public class CommonValue {
         public static final String ML_MODEL_GROUP_INDEX = ".plugins-ml-model-group";
         public static final String ML_MODEL_INDEX = ".plugins-ml-model";
         public static final String ML_TASK_INDEX = ".plugins-ml-task";
-        public static final Integer ML_MODEL_GROUP_INDEX_SCHEMA_VERSION = 2;
+        public static final Integer ML_MODEL_GROUP_INDEX_SCHEMA_VERSION = 3;
```
Comment (dbwiddis) [inline common/src/main/java/org/opensearch/ml/common/CommonValue.java:61]: Note to self as a reminder.  When we eventually get this on main we'll have to make sure there aren't any other conflicting schema updates.

File: common/src/main/java/org/opensearch/ml/common/CommonValue.java [hunk 2]
```diff
@@ -86,55 +86,56 @@ public class CommonValue {
...
+                + "    \"" + MLModelGroup.BACKEND_ROLES_FIELD + "\": {\n"
+                + "      \"type\": \"text\",\n"
+                + "      \"fields\": {\n"
+                + "        \"keyword\": {\n"
+                + "          \"type\": \"keyword\",\n"
+                + "          \"ignore_above\": 256\n"
+                + "        }\n"
+                + "      }\n"
+                + "    },\n"
+                + "    \"" + TENANT_ID + "\" : {\"type\": \"keyword\"},\n"
```
Comment (dbwiddis) [inline common/src/main/java/org/opensearch/ml/common/CommonValue.java:120]: This works 'cause it's valid JSON, but all the other strings (see `MlModelGroup.ACCESS` right below) do the "pretty" JSON equivalent.  Might want to be consistent.

**Review by dbwiddis — APPROVED** [review]
LGTM with a nit.

File: common/src/main/java/org/opensearch/ml/common/CommonValue.java (hunk 1 above)
Comment (dhrubo-os) [reply common/src/main/java/org/opensearch/ml/common/CommonValue.java:61]: +1

File: common/src/main/java/org/opensearch/ml/common/CommonValue.java (hunk 2 above)
Comment (dhrubo-os) [reply common/src/main/java/org/opensearch/ml/common/CommonValue.java:120]:
```
{
    "took": 11,
    "timed_out": false,
    "_shards": {
        "total": 1,
        "successful": 1,
        "skipped": 0,
        "failed": 0
    },
    "hits": {
        "total": {
            "value": 1,
            "relation": "eq"
        },
        "max_score": 1.0,
        "hits": [
            {
                "_index": ".plugins-ml-model-group",
                "_id": "k7BBoJEBmupm26gNRc4D",
                "_score": 1.0,
                "_source": {
                    "name": "Cohere embedding normal",
                    "latest_version": 1,
                    "description": "embedding sagemaker model",
                    "access": "public",
                    "created_time": 1724970517643,
                    "last_updated_time": 1724970517930,
                    "tenant_id": "1234567"
                }
            }
        ]
    }
}
```

This is mapping I got after making this change. Seems like similar?

**Pull request merged**